### PR TITLE
[ MO ][ RESHAPE ] Fixes in RNNSequenceNormalize transformation 

### PR DESCRIPTION
**Pull request by yekruglov**
Description: Fixes in implementation make this transformation reshape-able

JIRA: 34207

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests N/A: no new operations were enabled
* [x]  Transformation tests
* [x]  e2e model test with an update of ./tests/e2e_oss/utils/reshape_utils.py N/A: no new models were enabled
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list N/A: no new operations were enabled
* [x]  Supported **public** models list N/A: no new models were enabled
* [x]  New operations specification N/A: no new operations were enabled
* [x]  Guide on how to convert the **public** model N/A: no new models were enabled
* [x]  User guide update N/A

### CODE REVIEW

**Review by jane-intel — APPROVED** [review]

File: model-optimizer/extensions/middle/RNNSequenceNormalizeToIE.py [hunk 1]
```diff
@@ -154,14 +158,14 @@ def unsqueeze_num_directions(graph: Graph, match: dict):
...
 
-            reshape = Reshape(graph, dict(dim=old_shape))
+            unsqueeze = Unsqueeze(graph, dict())
 
-            reshape_dim_data = Const(graph, {'name': rnn_layer.name + '/SqueezeNumDirections/{}/Dim'.format(i),
-                                             'value': old_shape}).create_node_with_data()
-            reshape.create_node_with_data([data, reshape_dim_data],
-                                          dict(name=rnn_layer.name + '/SqueezeNumDirections/{}'.format(i)),
-                                          data_nodes=[old_data_node])
+            reshape_dim_data = Const(graph, {'name': rnn_layer.name + '/UnsqueezeNumDirections/{}/Dim'.format(i),
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/middle/RNNSequenceNormalizeToIE.py:163]: `reshape_dim_data` -> `unsqueeze_dim_data`?

File: model-optimizer/extensions/middle/RNNSequenceNormalizeToIE.py [hunk 2]
```diff
@@ -154,14 +158,14 @@ def unsqueeze_num_directions(graph: Graph, match: dict):
             graph.remove_edge(rnn_layer.id, old_data_node.id)
             graph.add_edge(rnn_layer.id, data.id, key=0, out=i)
 
-            reshape = Reshape(graph, dict(dim=old_shape))
+            unsqueeze = Unsqueeze(graph, dict())
 
-            reshape_dim_data = Const(graph, {'name': rnn_layer.name + '/SqueezeNumDirections/{}/Dim'.format(i),
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/middle/RNNSequenceNormalizeToIE.py:159]: use `rnn_layer.soft_get('name', rnn_layer.id)` instead of `rnn_layer.name`.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]
Can you provide a results of e2e run where the modified transformation was called?
How about transformation test?

**Comment by jane-intel** [comment]
>> Can you provide a results of e2e run where the modified transformation was called?
@lazarevevgeny, there is MXNET_RNN_Bidirectional_transducer_encoder and it fails here.

@yekruglov, please debug it

File: model-optimizer/extensions/middle/RNNSequenceNormalizeToIE.py [hunk 3]
```diff
@@ -154,14 +159,14 @@ def unsqueeze_num_directions(graph: Graph, match: dict):
...
-                                             'value': old_shape}).create_node_with_data()
-            reshape.create_node_with_data([data, reshape_dim_data],
-                                          dict(name=rnn_layer.name + '/SqueezeNumDirections/{}'.format(i)),
-                                          data_nodes=[old_data_node])
+            unsqueeze_dim_data = Const(graph, {'name': rnn_layer.name + '/UnsqueezeNumDirections/{}/Dim'.format(i),
+                                             'value': [direction_dim[i]]}).create_node_with_data()
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/middle/RNNSequenceNormalizeToIE.py:165]: It is not clear for me how the latest commit fixes the conversion issue of the MxNet model. Could you commen?

**Comment by lazarevevgeny** [comment]
@yekruglov , please, fix code style in nGraph. You can download the artefact with the patch and apply it.

**Comment by jane-intel** [comment]
> @yekruglov , please, fix code style in nGraph. You can download the artefact with the patch and apply it.
or run make style-apply

**Review by itikhono — APPROVED** [review]

**Comment by itikhono** [comment]
in general, LGTM, `prevalidate_and_infer `functions of Cells ops will be deleted in my next PR

**Review by jane-intel — APPROVED** [review]

**Comment by lazarevevgeny** [comment]
pre-commit failed once again. Please, retry one more time

**Review by lazarevevgeny — APPROVED** [review]

**Comment by jane-intel** [comment]
I restarted failed job

**Pull request merged**